src/AlgoliaSearch/Client.php

```
@@ -57,6 +58,8 @@ function __construct($applicationID, $apiKey, $hostsArray = null, $options = arr
         foreach ($options as $option => $value) {
             if ($option == "cainfo") {
                 $this->cainfoPath = $value;
+            } elseif ($option == "curl-options" and is_array($value)) {
```

I think we need to throw an explicit error if the `curl-options` option is not an array; otherwise it will be very complex to debug.